coalib/settings/ConfigurationGathering.py

```diff
@@ -200,15 +200,47 @@ def find_user_config(file_path, max_trials=10):
 
 
 def get_config_directory(section):
+    """
+    Retrieves the configuration directory for the given section.
+
+    Given an empty section:
+
+    >>> section = Section("name")
+
+    The configuration directory is not defined and will therefore fallback to
+    the current directory:
+
+    >>> get_config_directory(section) == os.path.abspath(".")
+    True
+
+    If the ``files`` setting is given with an originating coafile, the directory
+    of the coafile will be assumed the configuration directory:
+
+    >>> section.append(Setting("files", "**", origin="/tmp/.coafile"))
+    >>> get_config_directory(section) == os.path.abspath('/tmp/')
+    True
+
+    However if its origin is already a directory this will be preserved (the
+    path mangling is needed for the example to work on windows as well):
+
+    >>> tmpdir = os.path.dirname(os.path.abspath("/tmp")+ "/")
```

Comment:
Could you print tmpdir here ? So that the reader can understand what it is better ? Maybe even do `os.isdir(tmpdir)` to stress your point ?